lib/src/request.dart

```diff
@@ -9,50 +10,49 @@ import 'package:api_utils/src/api_response.dart';
 
 @visibleForTesting
 void setHttpClientForTesting(http.Client client) {
-  assert(client != null);
   _client = client;
 }
 
 http.Client _client = http.Client();
 
-typedef FromJson<T> = T Function(Map<String, dynamic>);
+typedef FromJson<T> = T Function(Map<String, dynamic>?);
```

Comment:
At first glance I don’t see why the map should be nullable, but still reviewing.

Reply:
Perhaps because `jsonDecode` returns dynamic? Maybe we can detect that and handle.

Reply:
True, `jsonDecode` returns `dynamic`. I converted the Map to be null-safe, and it seems to be working okay. We probably just to handle the case when the json returned is null for what ever reason.

Reply:
Since _handleResult is in our try catch, lets make the Map non nullable. If it does return null for some reason at least we will provide a solid error message.